torch/jit/frontend.py

```diff
@@ -346,17 +355,25 @@ def build_param_list(ctx, py_args, self_name):
             if arg is not None:
                 ctx_range = build_expr(ctx, arg).range()
                 raise NotSupportedError(ctx_range, _vararg_kwarg_err)
-    result = [build_param(ctx, arg, self_name, False) for arg in py_args.args]
-    result += [build_param(ctx, arg, self_name, True) for arg in py_args.kwonlyargs]
+
+    # List of Tuple of args and type as inferred by profile directed typing
+    arg_and_types = [(arg, next(iter(pdt_arg_types[arg.arg])) if pdt_arg_types[arg.arg] else None) for arg in py_args.args]
+    arg_and_types_kwonlyargs = [(arg, next(iter(pdt_arg_types[arg.arg])) if pdt_arg_types[arg.arg] else None)
```

Comment:
Sorry I didn't catch this earlier. It makes sense for `analyze` to maintain a Dict[str, set] as `all_args`'s type. After consolidation, the set will always only contain a single value. Therefore, we should change its type to avoid awkward use of `next(iter(set))` here.

Reply:
Yeah, I am planning to address this in my clean up PR.